### DISCUSSION TITLE
Regex update for Node.js - LTS only?

### DISCUSSION THREAD

**Discussion by codepunkt** [renovatebot/renovate#13725] [Request Help] [answered]
We have a repository with multiple `Dockerfile`, some of which have a `NODE_VERSION` environment variable that is supposed to be updated by renovate. So far, so good. This works fine!

However, we want to not allow major version upgrades, except for a very specific file - which is why we have set up `packageRules` as shown below. This still works fine.

However, the `Dockerfile` we're talking about that is supposed to receive major Node.js version upgrades receives those, but is updated to unsafe/non-LTS versions. We currenly have a PR open that upgrades to `17.3.1`, which is not wanted.

We would only like to get LTS version upgrades, including major ones, e.g. once `18.x` LTS is released, it should upgrade from `16.x` LTS to `18.x` LTS.

This is our `renovate.json5` config:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base", ":prHourlyLimitNone", ":prConcurrentLimitNone"],
  "enabledManagers": ["regex"],
  "regexMangers": [
    {
      "fileMatch": ["(^|/)Dockerfile%"],
      "matchStrings": ["ENV NODE_VERSION (?<currentValue>.*?)\n"],
      "depNameTemplate": "node",
      "datasourceTemplate": "npm"
    }
  ],
  "packageRules": [
    {
      "matchUpdateTypes": ["major"],
      "enabled": false,
    },
    {
      "matchUpdateTypes": ["major"],
      "enabled": true,
      "matchManagers": ["regex"],
      "matchPaths": ["nodejs/all/build/Dockerfile"],
    }
  ]
}
```

Is there any way this can be done?

**Comment by rarkins** [comment, marked as answer]
Try using the `node` datasource instead of `npm`. With `npm` I think you're looking up https://www.npmjs.com/package/node

**Reply by codepunkt** [reply]
That works. Thank you, Rhys!
